generators/pkg/generator/generate_test.go

```diff
@@ -6,33 +6,169 @@ import (
 	"github.com/google/go-cmp/cmp"
 	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestGenerateTask(t *testing.T) {
-	want := v1beta1.Task{
+	want := &v1beta1.Task{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "Task",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "github-build",
+			Labels: map[string]string{"generator.tekton.dev": "github-build"},
+		},
 		Spec: v1beta1.TaskSpec{
-			Workspaces: []v1beta1.WorkspaceDeclaration{{Name: "input",
-				MountPath: "/input"}},
+			Workspaces: []v1beta1.WorkspaceDeclaration{
+				{
+					Name:      "input",
+					MountPath: "/input",
+				},
+			},
+			Steps: []v1beta1.Step{
+				{
+					Container: corev1.Container{
+						Name:    "build",
+						Image:   "gcr.io/kaniko-project/executor:latest",
+						Command: []string{"/kaniko/executor"},
+						Args: []string{"--context=dir://$(workspaces.input.path)/src",
+							"--destination=gcr.io/wlynch-test/kaniko-test",
+							"--verbosity=debug"},
+					},
+				},
+			},
+		},
+	}
+	github := &GitHub{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "github-build",
+		},
+		Spec: GitHubSpec{
+			URL: "https://github.com/wlynch/test",
 			Steps: []v1beta1.Step{{Container: corev1.Container{
 				Name:    "build",
 				Image:   "gcr.io/kaniko-project/executor:latest",
 				Command: []string{"/kaniko/executor"},
 				Args: []string{"--context=dir://$(workspaces.input.path)/src",
 					"--destination=gcr.io/wlynch-test/kaniko-test",
 					"--verbosity=debug"},
-			}}}}}
-	spec := GitHubSpec{
-		URL: "https://github.com/wlynch/test",
-		Steps: []v1beta1.Step{{Container: corev1.Container{
-			Name:    "build",
-			Image:   "gcr.io/kaniko-project/executor:latest",
-			Command: []string{"/kaniko/executor"},
-			Args: []string{"--context=dir://$(workspaces.input.path)/src",
-				"--destination=gcr.io/wlynch-test/kaniko-test",
-				"--verbosity=debug"},
-		}}}}
-	got := GenerateTask(spec)
+			},
+			}}}}
+	got := GenerateTask(github)
 	if diff := cmp.Diff(want, got); diff != "" {
 		t.Errorf("Tasks mismatch (-want +got):\n %s", diff)
 	}
 }
+
+func TestGeneratePipeline(t *testing.T) {
+	ws := "source"
+	tasksName := []string{"fetch-git-repo", "build-from-repo", "final-set-status"}
+	want := &v1beta1.Pipeline{
```

Comment:
Is this supposed to match `testdata/pipeline.yaml`? As an alternative, you could read the result yaml from the testdata dir so you don't need to update multiple things for any test changes. (this is optional, feel free to push back or defer to another change if you want).